lib/helper/Polly.js

```diff
@@ -0,0 +1,135 @@
+const Helper = require('../helper');
+const requireg = require('requireg');
+const { appendBaseUrl } = require('../utils');
+
+let PollyJS;
+
+class Polly extends Helper {
+  constructor(config) {
+    super(config);
+    this._setConfig(config);
+    PollyJS = requireg('@pollyjs/core').Polly;
+  }
+
+  static _checkRequirements() {
+    try {
+      requireg('@pollyjs/core');
+    } catch (e) {
+      return ['@pollyjs/core@^2.5.0'];
+    }
+  }
+
+  /**
+   * Start mocking network requests/responses
+   *
+   * ```js
+   * I.startMocking('Backend APIs');
+   * ```
+   */
+  async startMocking(title = 'Test') {
+    if (!this.helpers && !this.helpers.Puppeteer) {
+      throw new Error('Puppeteer is the only supported helper right now');
+    }
+    return this._connectPuppeteer(title);
+  }
+
+  /**
+   * Connect Puppeteer helper to mock future requests.
+   */
+  async _connectPuppeteer(title) {
+    const adapter = require('@pollyjs/adapter-puppeteer');
+
+    PollyJS.register(adapter);
+    const { page } = this.helpers.Puppeteer;
+    await page.setRequestInterception(true);
+
+    this.polly = new PollyJS(title, {
+      adapters: ['puppeteer'],
+      adapterOptions: {
+        puppeteer: { page },
+      },
+    });
+  }
+
+  /**
+   * Mock response status
+   *
+   * ```js
+   * I.mock('GET', '/api/users', 200);
+   * I.mock('ANY', '/secretsRoutes/*', 403);
+   * I.mock('POST', '/secrets', { secrets: 'fakeSecrets' });
+   * ```
+   *
+   * Multiple requests
+   *
+   * ```js
+   * I.mock('GET', ['/secrets', '/v2/secrets'], 403);
+   * ```
+   */
+  async mock(method, oneOrMoreUrls, dataOrStatusCode) {
+    this._checkIfMockingStarted();
+    const handler = await this._getRouteHandler(
+      method,
+      oneOrMoreUrls,
+      this.options.url,
+    );
+
+    if (typeof dataOrStatusCode === 'number') {
+      const statusCode = dataOrStatusCode;
+      return handler.intercept((_, res) => res.sendStatus(statusCode));
+    }
+    const data = dataOrStatusCode;
+    return handler.intercept((_, res) => res.send(data));
+  }
+
+  /**
+   * Checks whether mocking is started.
+   */
+  _checkIfMockingStarted() {
+    if (!this.polly || !this.polly.server) {
```

Comment:
Maybe if mocking is not started, can we start it automatically? 
If (!this.polly) this.startMocking() ?
This can be a nice addition

Reply:
You mean, whenever user directly use `I.mockRequest()` we'll start internally?

Reply:
Ok, I've changed to start mocking automatically when `I.mockRequest()` used.